Roachtest azure firewall ports

smg260 opened the pull request:
roachprod: refactor azure security rules and open ports

This now covers most tests. 

roachtest: apt-get update for asyncpg


Epic: CC-25185
Release note: none

Commit:
roachprod: refactor azure security rules and open ports

This adds some more ports required for various roachtests and
refactors their creation with a helper function.

Epic: CC-25185

Release note: None

Commit:
roachtest: apt-get update for asyncpg

This is required so that packages can be found in azure.

Epic: CC-25185

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111927)
<!-- Reviewable:end -->

Commit:
roachprod: set azure default location westus2

`westus` does not support availability zones, but
`westus2` does. Over 20 roachtests use az.

Epic: CC-25185
Release note: none

smg260 commented:
combined with #111926
